File: LicenseManagementApi/Models/Entities/Customer.cs
namespace LicenseManagementApi.Models.Entities;

public class Customer
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Organization { get; set; }
    public bool IsVisible { get; set; } = true;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public ICollection<License> Licenses { get; set; } = new List<License>();
}

File: vue-ui-frontend/src/types/customer.ts
export interface Customer {
  id: string
  name: string
  email: string
  organization: string
  isVisible: boolean
  createdAt: string
  updatedAt: string
}

export interface CreateCustomerDto {
  name: string
  email: string
  organization: string
  isVisible: boolean
}

export interface UpdateCustomerDto {
  name: string
  email: string
  organization: string
  isVisible: boolean
}
